docs/api-gateway.md

```diff
@@ -91,3 +92,228 @@ Note that this way we don't need a Docker volume, which might work better in you
 $ curl localhost:4000
 Hi, I'm a demo service!
 ```
+
+## Dynamic URL parameters
+
+It's a common case that you want to fetch detail for some entity e.g. `/books/123`. To make sure the dynamic value `123` is correctly matched and forwarded API endpoint can be configured like this:
+
+```json
+[{
+  "id": "my-service",
+  "version_data": {
+    "default": {
+      "endpoints": [{
+        "id": "my-detail-endpoint",
+        "method": "GET",
+        "path": "/books/{book_id}"
+      }]
+    }
+  },
+  ...
+}]
+```
+
+Dynamic values in `path` have to be wrapped in curly braces. Value inside curly braces is up to you.
+
+## Publishing to event streams
+
+As an alternative to standard HTTP to HTTP communication, we offer a way how to produce event to Kafka/Kinesis via HTTP request. This means, that frontend can still send regular HTTP request and RIG will publish content of this request to Kafka topic or Kinesis stream.
+
+In this case, client receives response immediately (fire and forget).
+
+Configuration of such API endpoint might look like this:
+
+```json
+[{
+  ...
+  "version_data": {
+    "default": {
+      "endpoints": [{
+        "id": "my-endpoint",
+        "method": "POST",
+        "path": "/",
+        "target": "kafka"
+      }]
+    }
+  },
+  ...
+}]
+```
+
+> Important field is `target`. `target` describes what event stream to use. Currently it's either Kafka or Kinesis. HTTP method and `path` name are up to your choice.
+
+This type of endpoint requires specific HTTP request body:
+
+```json
+{
+  "event": {
+    "id": "069711bf-3946-4661-984f-c667657b8d85",
+    "type": "com.example",
+    "time": "2018-04-05T17:31:00Z",
+    "specversion": "0.2",
+    "source": "/cli",
+    "contenttype": "application/json",
+    "data":{
+      "foo": "bar"
+    }
+  },
+  "partition": "your_partition_key"
+}
+```
+
+As you can see, it needs 2 fields -- `event` and `partition`. `event` holds event itself (have to comply with Cloud events spec) and `partition` describes partition key to use.
+
+Topic/stream configuration is handled by environment variables. See `PROXY_KAFKA_*` and `PROXY_KINESIS_*` variables in [Operator's Guide](./rig-ops-guide.md).
+
+### Wait for response
+
+Publishing to event stream is asynchronous action (it's not request/response mechanism like HTTP is). Endpoint above can be configured, so that RIG waits for different event with the same correlation ID as the published event had. If such event is consumed, HTTP process is notified and client gets back response, otherwise it's timeout. The correlation ID is attached to Cloud events extension in published event called `rig`. In your backend systems you have to make sure that this extension field will be included also in the event that should be consumed by RIG and finish the process.
```

Comment:
```suggestion
Sometimes it makes sense to provide a simple request-response API to something that runs asychronously on the backend. For example, let's say there's a ticket reservation process that takes 10 seconds in total and involves three different services that communicate via message passing. For an external client, it may be simpler to wait 10 seconds for the response instead of polling for a response every other second.

A behavior like this can be configured using an endpoints' `response_from` property. When set to `kafka`, the response to the request is not taken from the `target` (e.g., for `target` = `http` this means the backend's HTTP response is ignored), but instead it's read from a Kafka topic. In order to enable RIG to correlate the response from the topic with the original request, RIG adds a correlation ID to the request (using a query parameter in case of `target` = `http`, or backed into the produced CloudEvent otherwise). Back-end services that work with the request need to include that correlation ID in their response; otherwise, RIG won't be able to forward it to the client (and times out).
```